google-cloud-bigtable-stats/src/main/java/com/google/cloud/bigtable/stats/StatsRecorderWrapper.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.stats;
+
+import com.google.api.core.InternalApi;
+import com.google.api.gax.tracing.ApiTracerFactory.OperationType;
+import com.google.api.gax.tracing.SpanName;
+import io.opencensus.stats.MeasureMap;
+import io.opencensus.stats.StatsRecorder;
+import io.opencensus.tags.TagContext;
+import io.opencensus.tags.TagContextBuilder;
+import io.opencensus.tags.TagKey;
+import io.opencensus.tags.TagValue;
+import io.opencensus.tags.Tagger;
+import io.opencensus.tags.Tags;
+import java.util.Map;
+
+/** A wrapper to record built-in metrics */
+@InternalApi("For internal use only")
+public class StatsRecorderWrapper {
+
+  private final OperationType operationType;
+
+  private final Tagger tagger;
+  private final StatsRecorder statsRecorder;
+  private final TagContext parentContext;
+  private final SpanName spanName;
+  private final Map<String, String> statsAttributes;
+
+  private MeasureMap measureMap;
+
+  public StatsRecorderWrapper(
+      OperationType operationType,
+      SpanName spanName,
+      Map<String, String> statsAttributes,
+      StatsWrapper statsWrapper) {
```

Comment:
I think you should make StatsWrapper be the factory for StatsRecorderWrapper

```java
class StatsWrapper {
  StatsRecorderWrapper createRecorder(OperationType operationType,
      SpanName spanName,
      Map<String, String> statsAttributes) {
     return new StatsRecorderWrapper(operationType, spanName, statsAttributes, statsRecorder, tagger);
 }
}

StatsRecorderWrapper(OperationType operationType,
      SpanName spanName,
      Map<String, String> statsAttributes,
     StatsRecorder recorder,
     Tagger tagger) 
```

This will better decouple the 2 and make it so that the child doesnt have to know about the parrent and be tested easier